packages/arb-token-bridge-ui/src/util/withdrawals/helpers.ts

```diff
@@ -54,83 +49,128 @@ export async function mapETHWithdrawalToL2ToL1EventResult(
     symbol: 'ETH',
     outgoingMessageState,
     decimals: 18,
-    l2TxHash: event.l2TxHash || event.transactionHash
+    l2TxHash: event.l2TxHash || event.transactionHash,
+    failed
   }
 }
 
+async function isWithdrawalFailed({
+  event,
+  l1Provider,
+  l2Provider,
+  l2ChainId
+}: {
+  event: L2ToL1EventResult & { l2TxHash?: string; transactionHash?: string }
+  l1Provider: Provider
+  l2Provider: Provider
+  l2ChainId: number
+}) {
+  const cacheKey = getL2ToL1MessageCacheKey({
+    event,
+    l2ChainId
+  })
+
+  // cached status is failure
+  const failedMessagesCache = JSON.parse(
+    localStorage.getItem(FAILED_MESSAGES_LOCAL_STORAGE_KEY) || '{}'
+  )
+  if (failedMessagesCache[cacheKey] === true) {
+    return true
+  }
+
+  const messageReader = new L2ToL1MessageReader(l1Provider, event)
+
+  // no cached status for this tx
+  // we check if the transaction has failed
+  if (typeof failedMessagesCache[cacheKey] === 'undefined') {
+    try {
+      await messageReader.getFirstExecutableBlock(l2Provider)
```

Comment:
Meanwhile I'm trying to figure out if we can call a less RPC heavy method to get this error message